### PR TITLE
Add '@' netgroup functionality

### PR DESCRIPTION
**Pull request by abergmann**
The openSUSE/SUSE net-snmp v5.7 package had a feature to use netgroups defined in `/etc/netgroup` or NIS/LDAP to gain access to a SNMP host. This feature was ported to our v5.9 and I would like to get it into the master branch.

How this feature works:
```
/etc/netgroup
nodes (node1,,) (node2,,) (node3,,)

/etc/snmp/snmpd.conf
rocommunity public @nodes
```

This makes it easier to define host groups that should have access to the SNMP host.

### CODE REVIEW

File: snmplib/transports/snmpUDPDomain.c [hunk 1]
```diff
@@ -454,20 +458,42 @@ netsnmp_udp_parse_security(const char *token, char *param)
...
+            while (getnetgrent(&host, &user, &domain)) {
+                /* Parse source address and network mask for each netgroup host. */
+                if (netsnmp_udp_resolve_source(host, &network, &mask) == 0) {
+                    /* Create a new com2Sec entry. */
+                    rc = netsnmp_udp_com2SecEntry_create(NULL, community, secName, contextName,
+                                                         &network, &mask, negate);
+                    netsnmp_udp_com2SecEntry_check_return_code(rc);
+                } else {
+                    config_perror("netgroup host address parsing issue");
+                    return;
```
Comment (bvanassche) [inline snmplib/transports/snmpUDPDomain.c:476]: Please make sure that endnetgrent() is called from this code path.

**Comment by abergmann** [comment]
@bvanassche are the changes dividing the proposed patch into 3 commits sufficient enough, or do I need to do further adjustments?

**Comment by bvanassche** [comment]
Thanks for the patches, and apologies for the delay.

**Comment by bvanassche** [comment]
I reworked some of the code in this pull request. Please help with reviewing commit d32c03d7e55cc4774b84586f2e17d981e7b45b9d.

**Pull request merged**